src/jobs_done10/server/app.py

```diff
@@ -190,6 +225,27 @@ def iter_jobs_done_requests_for_github_payload(
     )
 
 
+def verify_github_signature(headers: Dict[str, Any], data: bytes, secret: str) -> None:
+    """
+    Verify the post raw data and our shared secret validate against the signature in the header.
+
+    https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#delivery-headers
+    """
+    signature_name = "x-hub-signature-256"
+    try:
+        header_signature = headers[signature_name]
+    except KeyError:
+        raise SignatureVerificationError(f'Missing "{signature_name}" entry in headers')
+    algorithm = hmac.HMAC(secret.encode("UTF-8"), hashes.SHA256())
+    algorithm.update(data)
+    hash = algorithm.finalize().hex()
+    computed_signature = f"sha256={hash}"
+    if header_signature != computed_signature:
```

Comment:
There is an [`hmac` in the std library](https://docs.python.org/3.6/library/hmac.html#hmac.new).

Also, if you use [`cryptography.hazmat.primitives.hmac.HMAC.verify(signature)`](https://cryptography.io/en/latest/hazmat/primitives/mac/hmac/#cryptography.hazmat.primitives.hmac.HMAC.verify) it already raises an error.

It looks like giving info about the computed signature would be considered a security risk as an attacker could try to guess the secret.

Reply:
> There is an [hmac in the std library](https://docs.python.org/3.6/library/hmac.html#hmac.new).

Ahh cool thanks, will switch.

> It looks like giving info about the computed signature would be considered a security risk as an attacker could try to guess the secret.

Don't think so because the signature is put in the GitHub header anyway (in the open), and the algorithm used is public in the GitHub help, so in general I think it is harmless to show the signature. Regardless I will remove it as I don't think it helps anyway.

Reply:
The problem is not when they match, but when they don't, an attacker could send and bogus signature just to know what the calculated signature would look like and repeat the post with the right signature.
Or if he wanted he could brute force (or other attack) to discover the secret.

Also, the headers of the [request are encrypted by HTTPS](https://https.cio.gov/faq/#what-information-does-https-protect) (I was too lazy to look for an RFC) So the computed github headers signature will not go "in the open".

Reply:
Ahh OK thanks